cmd/kubeseal/main.go

```diff
@@ -586,25 +590,59 @@ func run(w io.Writer, secretName, controllerNs, controllerName, certURL string,
 		return nil
 	}
 
-	if !raw && !dumpCert {
+	var input io.Reader = os.Stdin
+	if inputFileName != "" {
+		f, err := os.Open(inputFileName)
+		if err != nil {
+			return nil
+		}
+		defer f.Close()
+
+		input = f
+	} else if !raw && !dumpCert {
 		if isatty.IsTerminal(os.Stdin.Fd()) {
 			fmt.Fprintf(os.Stderr, "(tty detected: expecting json/yaml k8s resource in stdin)\n")
 		}
 	}
 
+	// reEncrypt is the only "in-place" update subcommand. When the user only provides one file (the input file)
+	// we'll use the same file for output (see #405).
+	if reEncrypt && (outputFileName == "" && inputFileName != "") {
+		outputFileName = inputFileName
+	}
+	if outputFileName != "" {
+		// TODO(mkm): get rid of these horrible global variables
+		if ext := filepath.Ext(outputFileName); ext == ".yaml" || ext == ".yml" {
+			*outputFormat = "yaml"
+		}
+
+		var f *renameio.PendingFile
+		f, err = renameio.TempFile("", outputFileName)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err == nil {
+				f.CloseAtomicallyReplace()
```

Comment:
I believe it's worth adding a comment mentioning that this is preventing the output file from being written if the overall function result is an error.

Reply:
done